### PR TITLE
[Outreachy ] New dashboard design

### LINKED ISSUE #8832: [Outreachy] Implement New Dashboard Design
This issue is part of #8513

It will fix:

- [ ] Implement the new Dashboard design..refer to these [notes](https://github.com/publiclab/plots2/pull/8800#issuecomment-740052184)

### PR DESCRIPTION
**Pull request by RuthNjeri**
Fixes #8832
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

The comments marked as TODO will be addressed either within this PR or another after some clarification.

The current issue I have is determining what query fetches the featured topics, it does not seem like they are stored in a specific table?

Some Things To Note:

The topics card display notes, wikis and possibly other items. If a topic does not have any notes, I have included a text to keep note of this. See screenshots below.

Here are the screenshots:

Screenshot 2021-01-05 at 17 42 03
Screenshot 2021-01-05 at 17 42 09
Screenshot 2021-01-05 at 17 42 15
Screenshot 2021-01-05 at 17 42 28
Screenshot 2021-01-05 at 17 42 39

### CODE REVIEW

**Comment by RuthNjeri** [comment]
Hi @jywarren, I couldn't find your tag name on the list of reviewers I can tag. I'm tagging you on this comment instead.
Looking forward to your feedback 😄

File: app/controllers/subscription_controller.rb [hunk 1]
```diff
@@ -45,7 +45,9 @@ def add
         end
       end
 
-      case tag_selection_more_than_zero?(params[:tid])
+      # The route params does not have :tid, it only accepts the name..
+      # I think the name should be used to check if the user is subscribed to a tag already
+      case tag_selection_more_than_zero?(tag.tid)
```
Comment (RuthNjeri) [inline app/controllers/subscription_controller.rb:50]: The route params does not have :tid, it only accepts the name.

File: app/views/dashboard_v2/dashboard.html.erb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      <%= feature('dashboard-feature') %>
+      <%= render partial: "sidebar/dashboard_v2"%>
+    </div>
+  </div>
+  <%# Additional Topics %>
+  <div class="other-topics">
+    <h4>
+      <i class="form-grey d-none d-lg-inline"><%= translation('dashboard_v2.sidebar.interest_topics')%></i>
+    </h4>
+    <%# TODO Recommended topics %>
```
Comment (RuthNjeri) [inline app/views/dashboard_v2/dashboard.html.erb:21]: I am not sure which query to use to obtain recommended topics

File: app/views/dashboard_v2/dashboard.html.erb [hunk 3]
```diff
@@ -0,0 +1,87 @@
+<%# TODO: Topics to display if the user is new and not yet subscribed to other posts %>
```
Comment (RuthNjeri) [inline app/views/dashboard_v2/dashboard.html.erb:1]: The design showed that a new user will get an update saying "We have started you off with a few topics"
These topics should be featured. I currently don't know if there is a query I can use to show this

File: app/views/dashboard_v2/dashboard.html.erb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+  </div>
+  <%# Additional Topics %>
+  <div class="other-topics">
+    <h4>
+      <i class="form-grey d-none d-lg-inline"><%= translation('dashboard_v2.sidebar.interest_topics')%></i>
+    </h4>
+    <%# TODO Recommended topics %>
+    <br/>
+    <%# Trending topics %>
+    <small><a href="/tags" class="form-grey"><%= translation('dashboard_v2.sidebar.trending_topics') %>:</a></small>
```
[the remark this answers is not in the document]
Comment (jywarren) [reply app/views/dashboard_v2/dashboard.html.erb:24]: perhaps... but they are 2 diff. points of interaction - one at the top when you're building your subscriptions, and one where we 'catch' people having scrolled to the bottom. Let's see how it looks and maybe trim later? We should be able to use the same feature, so not too repetitive!

[the remark this answers is not in the document]
Comment (jywarren) [reply app/views/dashboard_v2/dashboard.html.erb:24]: oh sorry, trending vs. featured... but we can still re-use trending code!

File: app/views/dashboard_v2/dashboard.html.erb (hunk 2 above)
Comment (jywarren) [reply app/views/dashboard_v2/dashboard.html.erb:21]: this should be "featured" i believe!

File: app/views/dashboard_v2/dashboard.html.erb (hunk 3 above)
Comment (jywarren) [reply app/views/dashboard_v2/dashboard.html.erb:1]: Hi Ruth, as of the final discussion in https://github.com/publiclab/plots2/issues/6072#issuecomment-624275876, we are not pre-selecting any topics, so we can skip that portion of the welcome message. We should just include the blog, and rely on the bottom-of-the-column featured and trending topics to encourage new people to begin subscribing. Let's see how that looks and if it's exceptionally weird we can discuss further changes! But, for now the query isn't required. 

Or did you mean a query to see if the user is new? I believe there is a conditional on the current dashboard which we can use for this.

File: app/controllers/subscription_controller.rb (hunk 1 above)
Comment (jywarren) [reply app/controllers/subscription_controller.rb:48]: aha, ok - i'm fine with this! Uh, are we sure, as in, can we find if this code was tested by looking at the PR it was added, and skimming the functional and system tests to check? Will we need to modify any other code if we make this change? But, totally on-board with doing this, it makes sense!

**Comment by jywarren** [comment]
Hopefully was able to answer the questions you have! 

If we can address or dismiss remaining CodeClimate issues then we can merge this and do additional work in a follow-up PR, which is probably ideal! It's getting complex enough that it's a little harder to follow, so if we can do things in smaller batches that may make reviewing smoother and we think about it iteratively. 

Thanks, @RuthNjeri you're doing a fantastic job!

**Comment by jywarren** [comment]
Ah, and let's be sure to note the following from our discussion today in the planning issue for potentially breaking out into new issues, or at least discussing and making a decision on later:


* let's remove "empty" topics from dashboard, but ensure "follow" text at top and bottom of page links to /subscriptions.
* Another: should wiki edits or new wikis bump items to the top of the dashboard?
* Should topic cards show # of wikis?

@ebarry any thoughts on these last two?

**Comment by ebarry** [comment]
wow impressive awesome work!

File: app/controllers/subscription_controller.rb (hunk 1 above)
Comment (RuthNjeri) [reply app/controllers/subscription_controller.rb:48]: I did search it in the routes folder and didn't find it, I'll have a look at the test as well to confirm.

Comment (RuthNjeri) [reply app/controllers/subscription_controller.rb:48]: I've added a test for this, pushing it in a few. I also looked up the PR that was created for this change, no test was written and when I checked the routes folder, an add route with the params :tid was & is not present.

File: app/views/dashboard_v2/dashboard.html.erb (hunk 2 above)
Comment (RuthNjeri) [reply app/views/dashboard_v2/dashboard.html.erb:21]: On the mockup, it shows `recommended`.... should it be changed to featured?
Screenshot 2021-01-06 at 20 41 17

Comment (jywarren) [reply app/views/dashboard_v2/dashboard.html.erb:21]: Yes that sounds good. I think the mockup is just older and didn't get updated to the latest wording. Thanks!!

File: app/views/dashboard_v2/dashboard.html.erb (hunk 3 above)
Comment (RuthNjeri) [reply app/views/dashboard_v2/dashboard.html.erb:1]: Thanks for the clarification, it was mostly about the topics to display if it's a new user. 
About the `bottom-of-the-column featured and trending topics` should I remove any topics that a user has subscribed to from that section? Let's say that the user has subscribed to a topic "test" and test is either listed in the trending or featured topics, should it be removed, or can it stay the same?

File: app/views/dashboard_v2/dashboard.html.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (RuthNjeri) [reply app/views/dashboard_v2/dashboard.html.erb:24]: Yes! This makes sense!

**Comment by ebarry** [comment]
### Here's what staff talked about: 
- Should wiki edits or new wikis bump items to the top of the dashboard? **YES** When a wiki is added or edited, let’s bump up the topic card. Only display the name of the wiki in the list and not any kind of individual edit tracking
- Should topic cards show # of wikis? **YES** Let’s add number of wikis to the topic card!

### Here are other notes from our staff discussion: 
Mimi  28 minutes ago
can comments also count as activity and bump items to the top of the dashboard?

Mimi  2 minutes ago
From me: i’m wondering if there might be a way to integrate a bit more imagery on the dashboard? I love the tag pages right now and have gotten some really good feedback on how visually engaging they are. I’m excited to see the blog image and wonder if there’s a “what’s hot” or something that shows something like the cards on our current dashboard with an image for the post?

File: app/views/dashboard_v2/dashboard.html.erb (hunk 3 above)
Comment (jywarren) [reply app/views/dashboard_v2/dashboard.html.erb:1]: For trending topics i think we can remove already subscribed ones, but for featured, i think it'd be too tough to work with that if it's manually written... We'd have to use some brittle Javascript or something, and it's probably not worth it. 

Great!

**Comment by jywarren** [comment]
Thanks for reporting back, Liz! Regarding all of the staff feedback, let's be sure to move it all to its own issue and think on it in a later cycle of development, so as not to let it interrupt progress in this PR, sound good @RuthNjeri?

**Comment by RuthNjeri** [comment]
Thanks for the updates @ebarry.
Yes, I agree @jywarren, I can list all the TODOs and what Liz has mentions as separate issues. Also, I have addressed all of the feedback and added one more test for the `add` route. I think it will be ready to merge once all the tests have passed.

File: app/views/dashboard_v2/dashboard.html.erb (hunk 2 above)
Comment (RuthNjeri) [reply app/views/dashboard_v2/dashboard.html.erb:21]: Great! Thanks!

**Review by sagarpreet-chadha — COMMENTED** [review]
Great work so far !!!

**Review by Tlazypanda — COMMENTED** [review]
@RuthNjeri This looks amazing :tada: :100:

**Review by jywarren — APPROVED** [review]

**Comment by jywarren** [comment]
Looks perfect. Maybe later we may want to move the sidebar for the dashboard back into the dashboard folder? I don't really know if the sidebar views folder is really helping organization... but open to your suggestions!

Merging! Let's see this boot up on stable.publiclab.org!!!!

**Comment by jywarren** [comment]
Congrats!!!!!! 🎉 🎉 🎉 🎉 

thanks, everyone!!!

**Comment by cesswairimu** [comment]
Great job @RuthNjeri :tada: :tada: 

Looks like stable is broken :cry:

**Comment by RuthNjeri** [comment]
> Looks perfect. Maybe later we may want to move the sidebar for the dashboard back into the dashboard folder? I don't really know if the sidebar views folder is really helping organization... but open to your suggestions!
> 
> Merging! Let's see this boot up on stable.publiclab.org!!!!

@jywarren I totally agree. I was of the same opinion as well.. I'll add this as a task/issue

**Pull request merged**